event/src/main/java/jetbrains/jetpad/event/DerivedEvent.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2012-2016 JetBrains s.r.o
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.jetpad.event;
+
+public class DerivedEvent extends Event {
+  public final Event cause;
```

Comment:
In my opinion, by default we should use getters instead of fields. Especially if it's a public class of an open-source project. With getters we can instrument them if we want (additional behavior in the method). We can also override them if we want. With fields we lack all of this. And if we have lots of usages it could be more difficult to do refactorings especially if we have outside usages.